doc/sailfish-webview-qmozcontext.cpp

```diff
@@ -0,0 +1,367 @@
+    manifest file and a component referenced in the manifest. The would usually
+    also be a frontend QML file containing a \c WebView component which is
+    omitted for brevity.
+
+    The app \b main entry point which calls \c addComponentManifest.
+
+    \code
+        #include <QtQuick>
+        #include <sailfishapp.h>
+        #include <libsailfishwebengine/webengine.h>
+
+        int main2(int argc, char *argv[])
+        {
+            QGuiApplication *app = SailfishApp::application(argc, argv);
+            QQuickView *view = SailfishApp::createView();
+            view->setSource(SailfishApp::pathTo("application.qml"));
+
+            SailfishOS::WebEngine::instance() *webEngine = SailfishOS::WebEngine::instance();
+            webEngine->addComponentManifest(QLatin1String("/path/file.manifest"));
+
+            view->show();
+
+            return app->exec();
+        }
+    \endcode
+
+    The \b manifest file at /path/file.manifest
+
+    \code
+        component {20227a22-1722-4753-b8f6-c842b401b4c3} ExampleComponent.js
+        contract @mozilla.org/embedlite-example-component;1 {20227a22-1722-4753-b8f6-c842b401b4c3}
+        category app-startup ExampleComponent service,@mozilla.org/embedlite-example-component;1
+    \endcode
+
+    The javascript chrome \b component referenced in the manifest and stored
+    at /path/ExampleComponent.js
+
+    \code
+        const Ci = Components.interfaces;
+        const Cu = Components.utils;
+
+        Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+        Cu.import("resource://gre/modules/Services.jsm");
+
+        XPCOMUtils.defineLazyServiceGetter(Services, "embedlite",
+                                            "@mozilla.org/embedlite-app-service;1",
+                                            "nsIEmbedAppService");
+
+        Services.scriptloader.loadSubScript("chrome://embedlite/content/Logger.js");
+
+        function ExampleComponent() {
+          Logger.debug("JSComp: ExampleComponent.js loaded");
+        }
+
+        ExampleComponent.prototype = {
+          classID: Components.ID("{20227a22-1722-4753-b8f6-c842b401b4c3}"),
+
+          QueryInterface: XPCOMUtils.generateQI([Ci.nsIObserver]),
+
+          observe: function(aSubject, aTopic, aData) {
+            switch (aTopic) {
+            case "app-startup":
+              Services.obs.addObserver(this, "exampleTopic", false);
+              break;
+            case "exampleTopic":
+              Logger.debug("ExampleComponent: exapleTopic data: " + aData);
+              break;
+            }
+          },
+        };
+
+        this.NSGetFactory = XPCOMUtils.generateNSGetFactory([ExampleComponent]);
+    \endcode
+*/
+void QMozContext::addComponentManifest(const QString &manifestPath);
+
+/*!
+    \property QMozContext::initialized
+    \brief Whether the context has been initialized
+
+    This property returns true when the context has been initialized and false
```

Comment:
I think this sentence is unnecessary, personally.

Reply:
Yes, it is a bit. I've removed it.